### ISSUE TITLE
Duplicate dictionary keys are allowed if the keys are enum

### ISSUE THREAD

**Issue by ICatRegister** [godotengine/godot#50971]
### Godot version

3.3.2

### System information

Windows 10

### Issue description

Although duplicate dictionary keys are not allowed if the keys are constant, duplication does not cause compilation and runtime errors when using enumerations

### Steps to reproduce

This one is invalid:

```gdscript
const D: Dictionary = {
	1: "a",
	1: "b"
}
```

But this one is valid:

```gdscript
enum e {
	one = 1
}

const D: Dictionary = {
	e.one: "a",
	e.one: "b"
}
```

**Comment by kdiduk** [comment]
> Double-check if the issue is still present in master first, as it may be fixed already (or present in a different form), since GDScript was rewritten in master. 

In `master` it works correctly. I can reproduce this bug only in `3.x`. But I'm going to dig into it deeper anyway, so maybe I'll find something in `master` as well :slightly_smiling_face:

**Comment by kdiduk** [comment]
The PR #62574 with the fix of this issue is ready for review :angel:

**Comment by just-like-that** [comment]
This is also accepted by GDScript as being "valid" and takes the last representation of SEQUENCE being three.

```
enum Type {
	SEQUENCE = 01,
	COMBO    = 02,
	SEQUENCE = 03,
}
```

**Comment by lawnjelly** [comment]
Fixed by #62574.
